app/agents_v2/primary_agent/reasoning/schemas.py

```
@@ -0,0 +1,289 @@
+        if not phase_steps:
+            return 0.0
+        return sum(step.confidence for step in phase_steps) / len(phase_steps)
+    
+    def is_high_confidence(self, threshold: float = 0.8) -> bool:
+        """Check if overall reasoning confidence meets threshold"""
+        return self.overall_confidence >= threshold
+    
+    def get_reasoning_trace(self) -> str:
+        """Generate human-readable reasoning trace"""
+        trace_lines = []
+        trace_lines.append(f"# Reasoning Trace for Query Analysis")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)